airflow/api_fastapi/core_api/datamodels/dag_run.py

```diff
@@ -72,3 +74,35 @@ class DAGRunCollectionResponse(BaseModel):
 
     dag_runs: list[DAGRunResponse]
     total_entries: int
+
+
+class TriggerDAGRunPostBody(BaseModel):
+    """Trigger DAG Run Serializer for POST body."""
+
+    dag_run_id: str | None = None
+    logical_date: AwareDatetime | None = None
+    data_interval_start: AwareDatetime | None = None
+    data_interval_end: AwareDatetime | None = None
+
+    conf: dict | None = Field(default_factory=dict)
+    note: str | None
+
+    @model_validator(mode="after")
+    def check_data_intervals(cls, values):
+        if (values.data_interval_start is None) != (values.data_interval_end is None):
+            raise ValueError(
+                "Either both data_interval_start and data_interval_end must be provided or both must be None"
+            )
+        return values
+
+    @model_validator(mode="after")
+    def validate_dag_run_id(self):
+        if not self.dag_run_id:
+            self.dag_run_id = DagRun.generate_run_id(DagRunType.MANUAL, self.logical_date)
+        return self
+
+    # Mypy issue https://github.com/python/mypy/issues/1362
+    @computed_field  # type: ignore[misc]
+    @property
+    def _logical_date(self) -> datetime:
+        return self.logical_date or timezone.utcnow()
```

Comment:
Following @uranusjr, we remove everything related to `logical_date` (if the db allows us to insert like this), and we will leave a deeper layer populate that for us.